File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        res = ""

        for c in s :
            if c.isalpha() : res += c
            elif c == '#' : res += res
            elif c == '*' and len(res) > 0 : res = res[:-1]
            else : res = res[::-1]

        return res

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder res = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) res.append(c);
            else if (c == '#') res.append(res.toString());
            else if (c == '*' && res.length() > 0) res.setLength(res.length() - 1);
            else res.reverse();
        }

        return res.toString();
    }
}